File: backend/app/models/payload.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class PayloadGenerate(BaseModel):
    os_type: str
    shell_type: str
    lhost: str
    lport: int
    template: Optional[str] = None
    encode: bool = False


class PayloadResponse(BaseModel):
    payload: str
    template: Optional[str] = None
    os_type: str
    shell_type: str
    lhost: str
    lport: int


class PayloadHistoryEntry(BaseModel):
    id: int
    template: Optional[str] = None
    os_type: str
    shell_type: str
    lhost: str
    lport: int
    payload: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/payload.ts
export interface PayloadTemplate {
  id: string;
  name: string;
  os: string;
  shell_type: string;
  file?: string;
}

export interface PayloadGenerate {
  os_type: string;
  shell_type: string;
  lhost: string;
  lport: number;
  template?: string;
  encode: boolean;
}

export interface PayloadResponse {
  payload: string;
  template: string | null;
  os_type: string;
  shell_type: string;
  lhost: string;
  lport: number;
}

export interface PayloadHistoryEntry {
  id: number;
  template: string | null;
  os_type: string;
  shell_type: string;
  lhost: string;
  lport: number;
  payload: string;
  created_at: string;
}
